src/NuGet.Clients/NuGet.PackageManagement.UI/UserInterfaceService/NuGetUI.cs

```diff
@@ -54,7 +53,7 @@ public bool WarnAboutDotnetDeprecation(IEnumerable<NuGetProject> projects)
         {
             var result = false;
 
-            UIDispatcher.Invoke(() => { result = WarnAboutDotnetDeprecationImpl(projects); });
+            InvokeOnUIThread(() => { result = WarnAboutDotnetDeprecationImpl(projects); });
 
```

Comment:
I believe our analyzers don't work as well with helper methods like this one. I'd be surprised if you don't get VSTHRD010 warnings (now or in the future) for delegates like this when they invoke thread affinitized methods.
I tend to recommend people just use the full JTF.Run+SwitchToMainThread syntax for each use.

Reply:
yeah analyzers doesn't work well with helper methods, in this case there wasn't an explicit VS service call so it skips it. Is there a way to update the analyzer itself to work with helper methods?

Reply:
The latest version of the vs-threading analyzers _do_ let you [provide "additional files" to configure them](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/configuration.md), including defining methods that throw if not on the UI thread or guarantee to switch to the UI thread. But it's limited on the "switch" side to async methods that switch the caller, rather than take a delegate.
I'm hesitant to "fill" that hole because it's against our general guidance to use JTF directly everywhere you need it rather than use the helper method. The helper method reduces the pattern's length by only a line or two, but hides a great deal of very relevant data during a code review. For example, looking at this line, I have to ask myself:

1. Does `InvokeOnUIThread` return Task (and you're ignoring it)?
1. Does `InvokeOnUIThread` invoke the delegate synchronously or asynchronously?
1. Does `InvokeOnUIThread` use JTF inside or some older technology?
1. Does `InvokeOnUIThread` propagate exceptions from the delegate to the caller?

These are all very important questions to know the answers to during this code review. If you were using JTF directly, the familiar patterns bring the answers to mind immediately. But the helper method obfuscates all of it. If the line or two that the helper method saves you over JTF is that worthwhile to save, I encourage folks to let that incentivize them to write more pure-async code so they don't have to reference JTF.Run as often, since it tends to only show up at the boundaries.

Does that help?